frontend/src/Registration.jsx

```diff
@@ -0,0 +1,176 @@
+import {
+	Container,
+	TextField,
+	Grid,
+	Typography,
+	Button,
+} from "@material-ui/core";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+const validationSchema = yup.object().shape({
+	firstName: yup.string().required(),
+	lastName: yup.string().required(),
+	email: yup.string().email().required(),
+	birthYear: yup.number().positive().required(),
+	birthMonth: yup.number().min(1).max(12).required(),
+	birthDay: yup.number().min(1).max(31).required(),
+	username: yup.string().required(),
+	password: yup.string().required().min(8),
+});
+
+function Registration() {
+	const { register, handleSubmit, errors: validationErrors } = useForm({
+		resolver: yupResolver(validationSchema),
+	});
+
+	function onSubmit(validatedData, event) {
+		event.preventDefault();
+		event.target.reset();
+
+		const formattedRegistrationData = {
+			name: {
+				first: validatedData.firstName,
+				last: validatedData.lastName,
+			},
+			email: validatedData.email,
+			dateOfBirth: {
+				year: validatedData.birthYear,
+				month: validatedData.birthMonth,
+				day: validatedData.birthDay,
+			},
+			username: validatedData.username,
+			password: validatedData.password,
+		};
+
+		console.log(formattedRegistrationData);
```

Comment:
I guess this is the "stub" action where the actual action will take place? we should probably decide on how to handle this in our "in progress code". Maybe a TODO comment. Or a function call that is not implemented.

Reply:
Good catch! That's reasonable.

I can see "package-lock.json" at least in this branch.